hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/spark/sql/FileFormatUtilsForFileGroupReader.scala

```diff
@@ -124,4 +124,11 @@ object FileFormatUtilsForFileGroupReader extends SparkAdapterSupport {
       plan
     }
   }
+
+  def createStreamingDataFrame(sqlContext: SQLContext, relation: HadoopFsRelation, requiredSchema: StructType): DataFrame = {
+    val logRel = LogicalRelation(relation, isStreaming = true)
+    val resolvedSchema = logRel.resolve(requiredSchema, sqlContext.sparkSession.sessionState.analyzer.resolver)
+    Dataset.ofRows(sqlContext.sparkSession, applyFiltersToPlan(logRel, requiredSchema, resolvedSchema,
+      relation.fileFormat.asInstanceOf[HoodieFormatTrait].getRequiredFilters))
```

Comment:
Could the filters be passed when constructing `relation: HadoopFsRelation` instead of the post-processing here?

Reply:
It's counter-intuitive that the `relation` contains the filter but it's not applied.

Reply:
I don't know of a way.

```
case class HadoopFsRelation(
    location: FileIndex,
    partitionSchema: StructType,
    // The top-level columns in `dataSchema` should match the actual physical file schema, otherwise
    // the ORC data source may not work with the by-ordinal mode.
    dataSchema: StructType,
    bucketSpec: Option[BucketSpec],
    fileFormat: FileFormat,
    options: Map[String, String])(val sparkSession: SparkSession)
```

It's not really post processing though, because it still happens in the analysis stage. 

https://github.com/apache/hudi/pull/10225